controllers/front/result.php

```diff
@@ -0,0 +1,226 @@
+                        'paid_price' => $detail->getPaidPrice(),
+                        'currency' => $payment_currency,
+                        'total_refunded_amount' => 0,
+                        'created' => date('Y-m-d H:i:s'),
+                        'updated' => date('Y-m-d H:i:s'),
+                    );
+
+                    $dbFields = '`' . implode('`,`', array_keys($detail_arr)) . '`';
+                    $dbParams = "'" . implode("','", array_values($detail_arr)) . "'";
+
+                    $detail_query = "INSERT INTO `" . _DB_PREFIX_ . "iyzico_cart_detail` ({$dbFields}) VALUES ({$dbParams})";
+                    Db::getInstance()->execute($detail_query);
+
+                    $update_id_array = array(
+                        'order_id' => (int) $current_order_id,
+                        'updated' => date('Y-m-d H:i:s'),
+                    );
+
+                    Db::getInstance()->update('iyzico_api_log', $update_id_array, 'order_id = ' . (int) $cart->id);
+
+                    $update_array = array(
+                        'order_id' => (int) $current_order_id,
+                        'transaction_status' => 'success',
+                        'api_response' => pSQL($response->getRawResult()),
+                        'updated' => date('Y-m-d H:i:s'),
+                    );
+
+                    Db::getInstance()->update('iyzico_api_log', $update_array, 'id = ' . (int) $last_insert_id);
+                }
+            }
+            $this->context->smarty->assign(array(
+                'error' => $error_msg,
+                'total' => $total,
+                'currency' => $iso_code,
+                'locale' => $locale,
+            ));
+            $this->setTemplate('order_result.tpl');
+        } catch (\Exception $ex) {
+            $error_msg = $ex->getMessage();
+            $error_msg = !empty($error_msg) ? $error_msg : 'Some error occured.Please try again.';
+
+            $this->context->smarty->assign(array(
+                'error' => $error_msg,
+            ));
+            $this->setTemplate('order_result.tpl');
+        }
+    }
+
+    public function initPayment() {
+        $context = Context::getContext();
+        $cart = $context->cart;
+        $zero_total = $context->cookie->zero_total;
+
+        $iyzico = new Iyzicocheckoutform();
+        $currency = new Currency((int) ($cart->id_currency));
+        $iso_code = ($currency->iso_code) ? $currency->iso_code : '';
+        $cart_total = (float) $cart->getOrderTotal(true, Cart::BOTH);
+        $shipping_toal = (float) $cart->getOrderTotal(true, Cart::ONLY_SHIPPING);
+        $language_iso_code = $context->language->iso_code;
+        if ($cart_total == $shipping_toal && $zero_total) {
+            $total = 0;
+            $cart_total = 0;
+            $error_msg = ($language_iso_code == "tr") ? 'indirim tutarını alışveriş tutarınızı karşılamıştır.' : 'The amount of the discount amounting to meet your shopping.'  ;
```

Comment:
"indirim tutarını alışveriş tutarınızı karşılamıştır." yerine "Alışveriş tutarı indirim tutarına eşit olamaz."
"The amount of the discount amounting to meet your shopping." yerine "Cart total cannot be equal to discount amount"